### PR TITLE
feat: add support for secrets on workflow_call

### PR DESCRIPTION
**Pull request by glungley**
Approach to fix the secrets parts of #621

### CODE REVIEW

**Comment by glungley** [comment]
@npalm, seems you still need to approve if you're happy for the remaining checks to run

**Comment by npalm** [comment]
most likely you just need to run `yarn format`

**Comment by glungley** [comment]
@npalm could you merge this?

File: __tests__/fixtures/workflow/workflow.yml [hunk 1]
```diff
@@ -24,6 +24,12 @@ on:
           multiline description
         type: number
         required: true
+
+    secrets:
```
Comment (npalm) [inline __tests__/fixtures/workflow/workflow.yml:28]: Secrets are optional, I think it is better to add a dedicated workflow to check the secret parsing. This ensure workflows without secrets are also parsed correctly.

**Review by npalm — CHANGES_REQUESTED** [review]
Looks in general good, but I think we should ensure secrets are optional. Made two small comments.

File: src/action-docs.ts [hunk 2]
```diff
@@ -79,6 +81,7 @@ const inputOutputHeaders: Record<InputOutputType, string[]> = {
     "default",
   ],
   [InputOutputType.actionOutput]: ["name", "description"],
+  [InputOutputType.workflowSecret]: ["name", "description", "required"],
```
[the remark this answers is not in the document]
Comment (glungley) [reply src/action-docs.ts:84]: It is not documented as a separate entity, but it is in the example. This is the same for workflow inputs, where type is the only one called out individually and description is only mentioned in the example.

File: __tests__/fixtures/workflow/workflow.yml (hunk 1 above)
Comment (glungley) [reply __tests__/fixtures/workflow/workflow.yml]: Good point, updated

File: src/action-docs.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (glungley) [reply src/action-docs.ts:84]: I don't know what your preferred practise is, but I will leave conversations open and let you resolve them when you're happy with each.

[the remark this answers is not in the document]
Comment (npalm) [reply src/action-docs.ts:84]: Sorry description is part of the spec

**Review by npalm — APPROVED** [review]
@glungley thanks for the PR, and sorry for the long cycle time.

**Pull request merged**